compiler/src/iree/compiler/InputConversion/StableHLO/Preprocessing/FlattenTuplesInSCF.cpp

```diff
@@ -0,0 +1,266 @@
+                       ImplicitLocOpBuilder &b) {
+  auto tupleType = dyn_cast<TupleType>(oldType);
+  if (!tupleType) {
+    Value returnValue = *values->begin();
+    *values = {values->begin() + 1, values->end()};
+    return returnValue;
+  }
+
+  llvm::SmallVector<Value> subValues;
+  for (Type subType : tupleType.getTypes()) {
+    subValues.push_back(recursiveRetuple(subType, values, b));
+  }
+
+  return b.create<mlir::stablehlo::TupleOp>(tupleType, subValues).getResult();
+}
+
+void DetupleRegion(Region &srcRegion, Region &destRegion, ArrayRef<Type> types,
+                   IRMapping &mapping, ImplicitLocOpBuilder &b) {
+  auto beforeB = b.createBlock(&destRegion);
+  beforeB->addArguments(types, SmallVector<Location>(types.size(), b.getLoc()));
+  b.setInsertionPointToStart(beforeB);
+
+  llvm::SmallVector<Value> beforeResultVals;
+  for (auto argument : beforeB->getArguments()) {
+    beforeResultVals.push_back(argument);
+  }
+
+  llvm::ArrayRef<Value> beforeResults(beforeResultVals);
+  for (auto oldResult : srcRegion.front().getArguments()) {
+    auto newResult = recursiveRetuple(oldResult.getType(), &beforeResults, b);
+    mapping.map(oldResult, newResult);
+  }
+
+  b.setInsertionPointToEnd(&destRegion.front());
+  for (auto &srcop : srcRegion.front()) {
+    auto cloned = b.clone(srcop, mapping);
+    for (int i = 0; i < cloned->getNumResults(); i++) {
+      mapping.map(srcop.getResult(i), cloned->getResult(i));
+    }
+  }
+}
+
+class DetupleYieldOp : public OpRewritePattern<scf::YieldOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mlir::scf::YieldOp op,
+                                PatternRewriter &rewriter) const override {
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+    bool hasTuples = false;
+    IRMapping mapping;
+
+    llvm::SmallVector<Value> operands;
+    for (auto operand : op.getOperands()) {
+      hasTuples |= isa<TupleType>(operand.getType());
+      recursiveUntuple(operand, b, mapping, operands);
+    }
+
+    if (!hasTuples)
+      return rewriter.notifyMatchFailure(op, "no tupled arguments");
+
+    rewriter.replaceOpWithNewOp<mlir::scf::YieldOp>(op, operands);
+    return success();
+  }
+};
+
+class DetupleConditionOp : public OpRewritePattern<scf::ConditionOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mlir::scf::ConditionOp op,
+                                PatternRewriter &rewriter) const override {
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+    bool hasTuples = false;
+    IRMapping mapping;
+
+    llvm::SmallVector<Value> operands;
+    for (auto operand : op.getArgs()) {
+      hasTuples |= isa<TupleType>(operand.getType());
+      recursiveUntuple(operand, b, mapping, operands);
+    }
+
+    if (!hasTuples)
+      return rewriter.notifyMatchFailure(op, "no tupled arguments");
+
+    rewriter.replaceOpWithNewOp<mlir::scf::ConditionOp>(op, op.getCondition(),
```

Comment:
I think it's better to have consistent style. Some are using `scf::ConditionOp`, and some are using `mlir::scf::ConditionOp`. Can you unify the usage for all the patterns?